common/primitives/src/capacity.rs

```diff
@@ -13,6 +13,29 @@ pub trait TargetValidator {
 	fn validate(target: MessageSourceId) -> bool;
 }
 
+#[derive(
+	Clone, Copy, Debug, Decode, Encode, TypeInfo, Eq, MaxEncodedLen, PartialEq, PartialOrd,
+)]
+/// The type of staking a given Staking Account is doing.
+pub enum StakingType {
+	/// Staking account targets Providers for capacity only, no token reward
+	MaximumCapacity,
+	/// New reward program with new rules.
+	/// Defines a mechanism for locking tokens that is both time-sensitive and event-driven, with immutable thawing behavior.
+	CommittedBoost,
+	/// Staking account targets Providers and splits reward between capacity to the Provider
+	/// and token for the account holder
+	FlexibleBoost,
```

Comment:
note, we may flip the order of these depending on whether we're auto-migrating all existing boost stakes to the new staking type or not